pandas/core/indexing.py

```diff
@@ -1525,11 +1525,22 @@ def _getitem_axis(self, key, axis=0):
             # possibly convert a list-like into a nested tuple
             # but don't convert a list-like of tuples
             if isinstance(labels, MultiIndex):
+                if isinstance(key, ABCSeries):
+                    # GH 14730
+                    key = key.values.tolist()
+                elif isinstance(key, ABCDataFrame):
+                    # GH 15438
+                    raise NotImplementedError("Indexing a MultiIndex with a "
+                                              "DataFrame key is not "
+                                              "implemented")
+                elif hasattr(key, 'ndim') and key.ndim > 1:
```

Comment:
test using a numpy scalar as well (which is 0-dim), e.g. ``np.int64(5)`` (I think this works anyhow)